apps/server/src/modules/tool/uc/context-external-tool.uc.ts

```diff
@@ -30,6 +30,15 @@ export class ContextExternalToolUc {
 		return createdTool;
 	}
 
+	async deleteContextExternalTool(userId: EntityId, contextExternalToolId: EntityId): Promise<void> {
+		const tool = await this.contextExternalToolService.getContextExternalToolById(contextExternalToolId);
+		await this.ensureContextPermission(userId, tool.contextId, tool.contextType);
+
+		const promise: Promise<void> = this.contextExternalToolService.deleteContextExternalTool(tool);
```

Comment:
Two questions coming to my mind while reading: Why do you return the promise here and not just await the execution? Why do you state the type here but not in line 34? (IMO variables should not have type annotations if not required. Just my opinion, but as you seem undecided... ;))

Reply:
I forgot that in line 34 ;)